nx.json

```
@@ -0,0 +1,41 @@
+{
+  "$schema": "./node_modules/nx/schemas/nx-schema.json",
+  "affected": {
+    "defaultBase": "main"
+  },
+  "tasksRunnerOptions": {
+    "default": {
+      "runner": "nx/tasks-runners/default",
+      "options": {
+        "cacheableOperations": []
+      }
+    }
+  },
+  "targetDefaults": {
+    "check:dist": {
+      "dependsOn": ["^build"],
+      "options": {
+        "cwd": "{projectRoot}"
+      }
+    },
+    "check:types": {
+      "dependsOn": ["^build"]
```

It depends on its own build, am I right?
```suggestion
      "dependsOn": ["build"]
```

If you also want that dependencies are also built first, you can add:
```json
"build": {
      "dependsOn": ["^build"],
}
```